### PR TITLE
feat: switch component for OMP

### PR DESCRIPTION
**Pull request by gitanjli525**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
image

image

image

## Where to test it?

- [x] INTEG
- [x] SANDBOX
- [ ] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/APIUtils/APIUtils.res [hunk 1]
```diff
@@ -386,6 +386,11 @@ let useGetURL = () => {
         | None => `${userUrl}/${(userType :> string)->String.toLowerCase}`
         }
 
+      // O-M-P Switch
```
Comment (Riddhiagrawal001) [inline src/screens/APIUtils/APIUtils.res:389]: Can we rename it as Org-Merch-Profile List as there will be different routes for O-M-P switch

File: src/screens/Sidebar/Sidebar.res [hunk 2]
```diff
@@ -606,6 +616,9 @@ let make = (
           </div>
           <PinIconComponentStates isHSSidebarPinned setIsSidebarExpanded isSidebarExpanded />
         </div>
+        <RenderIf condition={featureFlagDetails.userManagementRevamp}>
```
Comment (Riddhiagrawal001) [inline src/screens/Sidebar/Sidebar.res:619]: Can we modify the condition to also check for totp feature flag as well ? If totp is not enabled, the context used to extract the organization, merchant, and profile IDs will fail.

File: src/screens/Sidebar/SidebarSwitch.res [hunk 3]
```diff
@@ -0,0 +1,32 @@
+module OrgMerchantSwitchCollapsed = {
+  @react.component
+  let make = () => {
+    let {orgId} = React.useContext(UserInfoProvider.defaultContext)
+    let {merchantId} = React.useContext(UserInfoProvider.defaultContext)
```
Comment (Riddhiagrawal001) [inline src/screens/Sidebar/SidebarSwitch.res:5]:
```suggestion
    let {merchantId, orgId} = React.useContext(UserInfoProvider.defaultContext)
```

**Comment by Riddhiagrawal001** [comment]
can you raise a separate PR for independent and core component changes . So that this PR only contains the business logic changes

File: src/screens/OMPSwitch/ProfileSwitch.res [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  let (fetchUpdatedProfileList, setFetchUpdatedProfileList) = React.useState(_ => true)
+  //   let {profileId} = React.useContext(UserInfoProvider.defaultContext)
+
+  let getProfileList = async () => {
+    try {
+      let url = getURL(~entityName=USERS, ~userType=#LIST_PROFILE, ~methodType=Get)
+      let response = await fetchDetails(url)
+      setProfileList(_ => response)
+    } catch {
+    | _ => ()
```
Comment (Riddhiagrawal001) [inline src/screens/OMPSwitch/ProfileSwitch.res:154]: Please handle the error cases

File: src/screens/OMPSwitch/OrgSwitch.res [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  let (orgList, setOrgList) = React.useState(_ => JSON.Encode.null)
+  let {orgId} = React.useContext(UserInfoProvider.defaultContext)
+
+  let getOrgList = async () => {
+    try {
+      let url = getURL(~entityName=USERS, ~userType=#LIST_ORG, ~methodType=Get)
+      let response = await fetchDetails(url)
+      setOrgList(_ => response)
+    } catch {
+    | _ => ()
```
Comment (Riddhiagrawal001) [inline src/screens/OMPSwitch/OrgSwitch.res:46]: Please handle the error cases

File: src/screens/OMPSwitch/MerchantSwitch.res [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+  let (showModal, setShowModal) = React.useState(_ => false)
+  let (fetchUpdatedMerchantList, setFetchUpdatedMerchantList) = React.useState(_ => true)
+
+  let getMerchantList = async () => {
+    try {
+      let url = getURL(~entityName=USERS, ~userType=#LIST_MERCHANT, ~methodType=Get)
+      let response = await fetchDetails(url)
+      setMerchantList(_ => response)
+    } catch {
+    | _ => ()
```
Comment (Riddhiagrawal001) [inline src/screens/OMPSwitch/MerchantSwitch.res:152]: Please handle the error cases

File: src/screens/Sidebar/SidebarSwitch.res [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    <div className="flex flex-col items-end gap-2">
+      <OrgSwitch />
+      <div className="flex">
+        <div className="w-6 h-10 border-blue-810 ml-10 border-dashed border-b border-l rounded" />
+        <MerchantSwitch />
+      </div>
+    </div>
+
+  <>
+    <RenderIf condition={isExpanded}> expandedContent </RenderIf>
```
Comment (PritishBudhiraja) [inline src/screens/Sidebar/SidebarSwitch.res:26]:
```suggestion
    <RenderIf condition={isExpanded}>
       <div className="flex flex-col items-end gap-2">
      <OrgSwitch />
      <div className="flex">
        <div className="w-6 h-10 border-blue-810 ml-10 border-dashed border-b border-l rounded" />
        <MerchantSwitch />
      </div>
    </div>
    </RenderIf>
```

Why to make seperate if no compuattional logic is there

Comment (gitanjli525) [reply src/screens/Sidebar/SidebarSwitch.res:26]: It looks clean

File: src/screens/OMPSwitch/OrgSwitch.res [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    getOrgList()->ignore
+    None
+  }, [])
+
+  let orgListArray =
+    orgList
+    ->getArrayFromJson([])
+    ->Array.map(item => {
+      item->getDictFromJsonObject->getString("org_id", "")
+    })
```
Comment (Riddhiagrawal001) [inline src/screens/OMPSwitch/OrgSwitch.res:56]: Please move this inside the getOrgList and make the state like selectbox options . Once name is present will use name to show in the dropdown . Set the state default as current org_id . 

do the same for merchant and profile as well

Comment (gitanjli525) [reply src/screens/OMPSwitch/OrgSwitch.res]: okay

File: src/screens/Sidebar/SidebarSwitch.res [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  }
+}
+
+@react.component
+let make = (~isExpanded=false) => {
+  let expandedContent =
+    <div className="flex flex-col items-end gap-2">
+      <OrgSwitch />
+      <div className="flex">
+        <div className="w-6 h-10 border-blue-810 ml-5 border-dashed border-b border-l rounded" />
```
Comment (Riddhiagrawal001) [inline src/screens/Sidebar/SidebarSwitch.res:20]:
```suggestion
        <div className="w-6 h-10 border-blue-810 ml-5 border-dashed border-b border-l rounded-bl-lg" />
```

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**